src/datum_logger.c

```diff
@@ -201,7 +201,7 @@ int datum_logger_queue_msg(const char *func, int level, const char *format, ...)
 	msg->calling_function[47] = 0;
 	msg->msg = &msg_buffer[buffer_id][msg_buf_idx[buffer_id]];
 	va_start(args, format);
-	i = vsnprintf(msg->msg, 1023, format, args);
+	i = vsnprintf(msg->msg, sizeof(msg->msg), format, args);
```

Comment:
`msg->msg` is a `char*`, so this will only be the size of a pointer and truncate all log messages...

Reply:
Ouch, it shows I didn't check the underlying type because I thought this is the place with a char[]. Sorry!

We can revert to 1023, especially because log_line that gets written to the file is 1200, so if we subtract about 80 bytes for the timestamp, function name and log level, we could fit about 1100 bytes, so that's fine.

**But I found something else here: the return value of vsnprintf is not checked and has an effect.**

In case of messages larger than 1023 it would waste a bit of the circular buffer. This is because when vsnprintf truncates it returns the **untruncated** length of the data. So, if it is >= 1023, then msg_buf_idx is moved by more than it really needs to, causing waste of the buffer. eg. 4000 instead of 1023:
```msg_buf_idx[buffer_id] += i+2; // moved by 4000 but we truncated to 1023```

If I am reading this right then I would suggest a quick check of the return value.
```
        i = vsnprintf(msg->msg, 1023, format, args);

        // clamp i to actual written value in order not to waste buffer space
        if (i >= 1023) {
                i = 1022;
        }
```
 Thoughts?